### ISSUE TITLE
Better way to selectively disable/enable CSRF protection on routes?

### ISSUE THREAD

**Issue by phanan** [laravel/framework#8165]
(Firstly, I've searched the repo, but there's no exact discussion/answer to the question it seems. Please just close this off if I'm wrong).

Now, Laravel 5 comes with CSRF protection enabled application-wide by default, which is good. If you want to disable it completely (RESTful?), just remove the middleware declaration from `app/Http/Kernel.php`, which is easy enough to do. The question is, how do you selectively disable it on several routes when still leaving it enabled at other places [1], and vice-versa [2]? 

From what I know, the current solution for [1] could be something like this:

``` php
// Single route
Route::post('/lorem/ipsum', 'LoremController@Ipsum', ['middleware' => 'csrf']);

// Grouped routes
Route::group(['middleware' => 'csrf'], function()
{
    Route::post('/', function() {});
    Route::post('user/profile', function() {});
});
```

But what about [2]? Apparently there's no such approach.  [This guy](http://www.techigniter.in/tutorials/disable-csrf-check-on-specific-routes-in-laravel-5/) came up with something like this:

``` php
public function handle($request, Closure $next)
{
    // disable CSRF check on following routes
    $skip = array(
                'user/path/xys',
                'user/profile',
                'my/unprotected/route'
                );

    foreach ($skip as $key => $route) {
        // skip csrf check on route
        if($request->is($route)) {
            return parent::addCookieToResponse($request, $next($request));
        }
    }

    return parent::handle($request, $next);
}

```

Basically he checks the URI of every request and compares it against a pre-defined list. This works, but the hard-coded paths make the whole thing look hacky and ugly, and HTTP verbs are not catered yet.

So, the question is, what is a better way to selectively disable CSRF on routes? Can/should Laravel introduce some elegant approach to allow disabling CSRF (and other middleware) on a per-route basis, just like we can enable them?

Again, sorry if this is something Laravel is already capable to do and I'm just babbling nonsense.

**Comment by RomainLanz** [comment]
Juste move the CSRF Middleware (`VerifyCsrfToken`) to `$routeMiddleware`.

https://github.com/laravel/laravel/blob/master/app/Http/Kernel.php

**Comment by phanan** [comment]
@RomainLanz Err... Can you elaborate a bit further? How does moving the middleware help with the issue?

**Comment by GrahamCampbell** [comment]
Ask on the forums please.

**Comment by alicompu** [comment]
Hi for the group paths that you want to apply to everyone, use this method, which is my 5.5 larval version. Use the star =>
go to app/Http/Middleware/VerifyCsrfToken and add

    protected $except = [
        '/user/*'
    ];
This is also my user's path

    Route::group(['prefix' => 'user', 'namespace' => 'User', 'as' => 'user.'] , function (){
